### PR TITLE
add the option to select acme challenge location and hook

### PR DESCRIPTION
**Pull request by melvinsoft**
This solves a bug introduced recently in Tahoe's custom domains.

Tahoe custom domains relies on GSCFuse to mount a Google Storage bucket in where the acme lets encrypt acme challenge is stored, and the generated cert as well. This allows to Tahoe to be able to generate and have the certs available in multiple app servers.

More info abut custom domains architecture: https://github.com/appsembler/tahoe/blob/master/docs/domains.md

Recently after the production push, custom domains stoped working, after some hours of debug with @bezidejni we found this: https://github.com/GoogleCloudPlatform/gcsfuse/issues/261

Looks like Google Cloud banned objects starting with `/.well-known/acme-challenge`. we create those every-time we need to generate q new cert for a custom domain.

The PR introduces the option to use a manual command to generate the cert, which allow us use a pre hook script, to store the acme challenge in a different path inside the GS bucket, and change the configuration in nginx to return the file content in the let's encrypt challenge.

I'd like to reviews from the @bezidejni and @thraxil first, to make sure that this is addresses the problem, and a Black team review to make sure that we con't break anything for customer sites.

### CODE REVIEW

**Comment by melvinsoft** [comment]
[(8) Cybereason: Error with custom domains](https://trello.com/c/8CDzfDs7/1954-8-cybereason-error-with-custom-domains)

**Comment by OmarIthawi** [comment]
@melvinsoft @thraxil Although this PR could solve the problem. I personally think it's adding some unnecessary complexity for smaller deployments.

 I've started a [conversation on Slack about that](https://github.com/appsembler/edx-configs/pull/284). Let's please discuss the issue further.

File: letsencrypt/files/authenticator.sh [hunk 1]
```diff
@@ -0,0 +1,2 @@
+#!/bin/bash
+echo $CERTBOT_VALIDATION > {{ letsencrypt_webroot }}/{{ letsencrypt_alternative_acme_folder }}/$CERTBOT_TOKEN
```
Comment (OmarIthawi) [inline letsencrypt/files/authenticator.sh]: `$CERTBOT_VALIDATION` -> `"$CERTBOT_VALIDATION"`

`$CERTBOT_TOKEN` -> `"$CERTBOT_TOKEN"`

Comment (melvinsoft) [reply letsencrypt/files/authenticator.sh]: Done!

**Comment by melvinsoft** [comment]
@OmarIthawi @thraxil Thanks for your reviews!

**Pull request merged**